src/components/PopupWithForm/PopupWithForm.js

```diff
@@ -1,18 +1,32 @@
 import React from "react";
+import "../PopupWithForm/PopupWithForm.css";
+import { useFormWithValidation } from '../../Hooks/useFormWithValidation';
+import { Link } from 'react-router-dom';
+
+function PopupWithForm({ isOpen, children, onClose, onSubmit, formTitle, buttonText, linkClickHandler, linkText }) {
+    const { isValid } = useFormWithValidation();
```

Comment:
`    const { isValid } = useFormWithValidation();
`

It won't work like that

* The submit button should be **disabled** if the form is invalid https://disk.yandex.ru/i/4R0j-VIPD-VVtA 

It means that you need to pass `isValid ` into the **props** of `PopupWithForm` from the components that use `PopupWithForm` and validation

Reply:
Done